contracts/farm-bot.sol

```diff
@@ -202,46 +206,49 @@ contract FarmBot is Owned, FarmbotERC20 {
 
         // compute bounty for the caller
         uint256[] memory _tokenBalances = new uint256[](rewardsTokens.length);
-	TokenAmount[] memory _interestAccrued = new TokenAmount[](rewardsTokens.length);
-
-	for (uint i=0; i< rewardsTokens.length; i++) {
-	    _tokenBalances[i] = rewardsTokens[i].balanceOf(address(this));
-	    _interestAccrued[i] = TokenAmount(rewardsTokens[i], _tokenBalances[i]);
-	}
-
 	uint256[] memory _bountyAmounts = new uint256[](rewardsTokens.length);
+	uint256[] memory _reserveAmounts = new uint256[](rewardsTokens.length);
+
         {  // block is to prevent 'stack too deep' compilation error.
-	    TokenAmount[] memory _feeBounties = revoBounty.calculateFeeBounty(_interestAccrued);
-	    for (uint i=0; i < _feeBounties.length; i++) {
-		_bountyAmounts[i] = _feeBounties[i].amount;
+	    TokenAmount[] memory _interestAccrued = new TokenAmount[](rewardsTokens.length);
+	    for (uint i=0; i< rewardsTokens.length; i++) {
+		_tokenBalances[i] = rewardsTokens[i].balanceOf(address(this));
+		_interestAccrued[i] = TokenAmount(rewardsTokens[i], _tokenBalances[i]);
+	    }
+
+	    TokenAmount[] memory _bountyFees = revoBounty.calculateBountyFee(_interestAccrued);
+	    TokenAmount[] memory _reserveFees = revoBounty.calculateReserveFee(_interestAccrued);
+	    require(_bountyFees.length == _reserveFees.length, "Got conflicting results from RevoBounty");
+	    for (uint i=0; i < _bountyFees.length; i++) {
+		_bountyAmounts[i] = _bountyFees[i].amount;
+		_reserveAmounts[i] = _reserveFees[i].amount;
 		require(_bountyAmounts[i] <= maxFeeNumerator * _tokenBalances[i] / maxFeeDenominator, "Bounty amount too high");
```

Comment:
```suggestion
		require(_bountyAmounts[i] + _reserveAmounts[i] <= maxFeeNumerator * _tokenBalances[i] / maxFeeDenominator, "Bounty amount too high");
```

Reply:
The goal of this check is to make sure the user has a guaranteed fee maximum that the owners can't change once the contract is deployed. We should include the reserve fee in this guarantee.

Reply:
Ahh of course; that's a great catch, thanks!